From bc6e8bfa1cd91ec3946b0bd57829ff9e6a733907 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 9 Feb 2017 16:40:55 -0800
Subject: [PATCH] TOOLS-1689 sdcadm hang on exit: suspect urclient leaving
 handles open after .close() + cmdln softexit Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 lib/cli/index.js | 6 ++++--
 package.json     | 2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/cli/index.js b/lib/cli/index.js
index 2fa2f1d..e98fbe1 100644
--- a/lib/cli/index.js
+++ b/lib/cli/index.js
@@ -467,8 +467,10 @@ module.exports = CLI;
 
 if (require.main === module) {
     var cli = new CLI();
-    cmdln.main(cli, process.argv, {
+    cmdln.main(cli, {
+        argv: process.argv,
         showCode: true,
-        showErr: true
+        showErr: true,
+        finale: 'exit'
     });
 }
diff --git a/package.json b/package.json
index bfa6be5..1630a4b 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "assert-plus": "^1.0.0",
     "backoff": "2.5.0",
     "bunyan": "0.23.1",
-    "cmdln": "4.2.0",
+    "cmdln": "4.2.1",
     "extsprintf": "^1.3.0",
     "kthxbai": "~0.4.0",
     "joyent-schemas": "git+https://github.com/joyent/schemas.git#385e6eb",
-- 
2.21.0

